packages/quicktype-core/src/language/Rust.ts

```diff
@@ -52,6 +52,64 @@ export const rustOptions = {
     leadingComments: new BooleanOption("leading-comments", "Leading Comments", true)
 };
 
+type nameToParts = (name: string) => string[];
+type partsToName = (parts: string[]) => string;
+type namingStyle = {
+    regex: RegExp,
+    toParts: nameToParts,
+    fromParts: partsToName,
+};
+
+const NamingStyle: { [key: string]: namingStyle } = {
+    "snake_case": {
+        regex: /^[a-z][a-z0-9]*(_[a-z0-9]+)*$/,
+        toParts: (name: string): string[] => name.split('_'),
+        fromParts: (parts: string[]): string => parts.map(p => p.toLowerCase()).join('_'),
+    },
+    "SCREAMING_SNAKE_CASE": {
+        regex: /^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$/,
+        toParts: (name: string): string[] => name.split('_'),
+        fromParts: (parts: string[]): string => parts.map(p => p.toUpperCase()).join('_'),
+    },
+    "camelCase": {
+        regex: /^[a-z]+([A-Z0-9][a-z]*)*$/,
+        toParts: (name: string): string[] => NamingStyle.snake_case.toParts(name.replace(/(.)([A-Z])/g, "$1_$2")),
+        fromParts: (parts: string[]): string => parts
+            .map((p, i) =>
+                i === 0
+                    ? p.toLowerCase()
+                    : p.substring(0, 1).toUpperCase() + p.substring(1).toLowerCase())
+            .join(''),
+    },
+    "PascalCase": {
+        regex: /^[A-Z][a-z]*([A-Z0-9][a-z]*)*$/,
+        toParts: (name: string): string[] => NamingStyle.snake_case.toParts(name.replace(/(.)([A-Z])/g, "$1_$2")),
+        fromParts: (parts: string[]): string => parts
+            .map(p => p.substring(0, 1).toUpperCase() + p.substring(1).toLowerCase())
+            .join(''),
+    },
+    "kebab-case": {
+        regex: /^[a-z][a-z0-9]*(-[a-z0-9]+)*$/,
+        toParts: (name: string): string[] => name.split('-'),
+        fromParts: (parts: string[]): string => parts.map(p => p.toLowerCase()).join('-'),
+    },
+    "SCREAMING-KEBAB-CASE": {
+        regex: /^[A-Z][A-Z0-9]*(-[A-Z0-9]+)*$/,
+        toParts: (name: string): string[] => name.split('-'),
+        fromParts: (parts: string[]): string => parts.map(p => p.toUpperCase()).join('-'),
+    },
+    "lowercase": {
+        regex: /^[a-z][a-z0-9]*$/,
+        toParts: (name: string): string[] => [name],
+        fromParts: (parts: string[]): string => parts.map(p => p.toLowerCase()).join(''),
+    },
+    "UPPERCASE": {
+        regex: /^[A-Z][A-Z0-9]*$/,
+        toParts: (name: string): string[] => [name],
+        fromParts: (parts: string[]): string => parts.map(p => p.toUpperCase()).join(''),
```

Comment:
You don't need to write the types for each toParts and fromParts wince you declared them for the entire struct.

Reply:
Actually I use them to detect the input fields formats.
I also use them to detect if a field does not match the global type.